test/subscription_map.hpp

```diff
@@ -405,20 +398,13 @@ class multiple_subscription_map
         }
 
         auto& subscription_set = h_iter->second.value;
-        auto subscription_iter = subscription_set.find(value);
-        if (subscription_iter != subscription_set.end()) {
-            auto hint = std::next(subscription_iter);
-            subscription_set.erase(subscription_iter);
-            subscription_set.insert(hint, std::move(value));
-        }
-        else{
-            subscription_set.insert(std::move(value));
-        }
+        bool insert_result = subscription_set.insert(std::move(value)).second;
+        return std::make_pair(h, insert_result);
     }
 
     // Remove a value at the specified subscription path
     // returns the value of the removed element (if found)
-    boost::optional<Value> remove(MQTT_NS::string_view subscription, Value const& value) {
+    size_t remove(MQTT_NS::string_view subscription, Value const& value) {
```

Comment:
I think that the name should be `eraseP too.

Reply:
i will fix also